src/doc/sprite.cpp

```diff
@@ -456,14 +456,10 @@ RgbMap* Sprite::rgbMap(const frame_t frame,
         return nullptr;
     }
   }
-  int maskIndex;
-  if (forLayer == RgbMapFor::OpaqueLayer)
-    maskIndex = -1;
-  else {
-    maskIndex = palette(frame)->findMaskColor();
-    if (maskIndex == -1)
-      maskIndex = 0;
-  }
+  int maskIndex = palette(frame)->findMaskColor();
+  maskIndex = (maskIndex == -1 ?
+                (forLayer == RgbMapFor::OpaqueLayer ? -1 : 0)
+                : maskIndex);
```

Comment:
It looks like the if-condition could be written as:

```c++
  int maskIndex = palette(frame)->findMaskColor();
  if (maskIndex == -1 &&
      forLayer == RgbMapFor::TransparentLayer) {
    maskIndex = 0;
  }
```

Anyway, searching for the source of these lines I saw that the previous version used the `transparentColor()` value which seems reasonable to get the mask color for a transparent layer in an indexed image:

https://github.com/aseprite/aseprite/commit/2785a9fef71ff8aa45aee086b9d30f8a8805f67f#diff-caddafaab7e9e69d1dbf807f44df633b27d21bedee0694ac3e706627b4b5c43cL398-L399

And that makes me think why it's not using the transparent color here in for that specific case now (and hard coding 0 instead).

There is other thing in that patch where the transparent color is set directly when we convert from indexed -> RGB to 0 (and not restored to its original value in case of undo):

https://github.com/aseprite/aseprite/commit/2785a9fef71ff8aa45aee086b9d30f8a8805f67f#diff-6967ed262db5d2bd98a11d2b60afef980cfb231b17be861e2acad05fdc0230c3R181-R186

It's calling `setTransparentColor()` instead of using a `cmd::SetTransparentColor`.

Tomorrow I'll try to dedicate some time merging this commit but then trying to find a solution to these new issues.